packages/tocco-util/src/intl/intl.js

```diff
@@ -24,16 +28,57 @@ const fetchOptions = {
   credentials: 'include'
 }
 
-const getUserInfo = () => (
-  fetch(`${__BACKEND_URL__}/nice2/username`, fetchOptions)
-    .then(response => response.json())
-)
+export const getUserLocale = async() => {
+  const cachedUserLocale = cache.get('user', 'locale')
+
+  if (cachedUserLocale) {
+    return cachedUserLocale
+  }
+
+  const response = await fetch(`${__BACKEND_URL__}/nice2/username`, fetchOptions)
+  const userInfo = await response.json()
+  const {locale} = userInfo
+  cache.add('user', 'locale', locale)
+  return locale
+}
+
+export const loadTextResources = async(locale, modules) => {
+  let result = {}
+  const notLoadedModules = []
+
+  modules.forEach(module => {
+    const cachedModuleResource = cache.get('textResource', module)
```

Comment:
what happens when the user changes his locale? the cache contains only text resources for locale, right? is the cache invalidated somewhere when the locale changes?

Reply:
not yet. Could be done in the future. For now the user has do close the tab and reopen the page.